### PR TITLE
docs: improve the README for examples/call_v_from_c/ with instructions for macos

### PR DESCRIPTION
**Pull request by wenq1**
Add macOS docs

### CODE REVIEW

File: examples/call_v_from_c/README.md [hunk 1]
```diff
@@ -37,3 +49,5 @@ Step 2: Compile the c file using `gcc test_print.c v_test_print.c -o test_print
 `gcc test_math.c v_test_math.c -o test_math -lgc -lm`.
 
 Step 3: Run the compiled c file using `./test_print` or `./test_math`.
+
+On MacOS, first do `brew install libgc` if using static build. When compiling/linking, add `-I/usr/local/include -L/usr/local/lib` (for x86_64) or `-I/opt/homebrew/include -L/opt/homebrew/lib` (for arm64)
```
Comment (spytheman) [inline examples/call_v_from_c/README.md:53]: This should be before line 19, right after the title `#### On Mac OSX:`, otherwise it is confusing.

Comment (wenq1) [reply examples/call_v_from_c/README.md:53]: No it should be in the ### Compile as a c file ### section. Please review and reconsider.

Comment (spytheman) [reply examples/call_v_from_c/README.md:53]: I did. Again, please do move the text about `brew install libgc` before line 19.

Comment (spytheman) [reply examples/call_v_from_c/README.md:53]: Following instructions out of order is confusing. Ignoring instructions that are not meant for your particular case is easy.

Comment (spytheman) [reply examples/call_v_from_c/README.md:53]: If you prefer, you can add a separate section `### Requirements`, that is before everything else, and then everything related to the requirements there, including the `brew install libgc` line.

Comment (wenq1) [reply examples/call_v_from_c/README.md]: Done.

**Pull request merged**